gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/CircularDependencyChecker.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.common.impl;
+
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.VersionDoesNotExistException;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.hadoop.fs.Path;
+
+
+public class CircularDependencyChecker {
+
+  /**
+   * @param cs - {@gobblin.config.store.api.ConfigStore} to check circular dependency
+   * @param uri - URI relative to the input ConfigStore. This is the starting point to check circular dependency
+   */
+  public static void checkCircularDependency(ConfigStore cs, String version, URI uri) throws VersionDoesNotExistException{
+    // check ancestor chain
+    checkAncestorCircularDependency(cs, version, uri, uri, new ArrayList<URI>());
+
+    // check self imported chain
+    checkImportCircularDependency(cs, version, uri, uri, new ArrayList<URI>());
+  }
+
+  /**
+   * 
+   * @param cs - ConfigStore to check with
+   * @param version - ConfigStore version to check with
+   * @param initialURI - the initial URI passed from public function checkCircularDependency(...)
+   * @param uri - current URI to check with
+   * @param previous - all previously imports
+   */
+  private static void checkAncestorCircularDependency(ConfigStore cs, String version, URI initialURI, URI uri,
+      List<URI> previous) throws VersionDoesNotExistException{
+    String pStr = uri.getPath();
+    if (pStr.length() == 0)
+      return;
+
+    Path p = (new Path(uri.getPath())).getParent();
```

Comment:
We shouldn't depend on Hadoop to do these operations. See http://stackoverflow.com/questions/10159186/how-to-get-parent-url-in-java on how to get parent uri.